src/components/Fund/FundDetails/FundDetailsContainer.js

```diff
@@ -53,94 +54,108 @@ const FundDetailsContainer = ({
-    mutator.ledger.reset();
-    mutator.allocatedFrom.reset();
-    mutator.allocatedTo.reset();
-    mutator.group.reset();
-    mutator.budgets.reset();
+  const [fetchFundResource, setFetchFundResource] = useState(false);
+  const [compositeFund, setCompositeFund] = useState({ fund: {}, groupIds: [] });
+  const [fundTypeName, setFundTypeName] = useState('');
+  const [ledger, setLedger] = useState({});
+  const [allocatedFromFunds, setAllocatedFromFunds] = useState('');
+  const [allocatedToFunds, setAllocatedToFunds] = useState('');
+  const [budgets, setBudgets] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
 
-    mutator.fund.GET().then(response => {
-      const {
-        fund: {
-          ledgerId,
-          fundTypeId,
-          allocatedFromIds,
-          allocatedToIds,
-        },
-      } = response;
-
-      if (fundTypeId) {
-        mutator.fundType.GET({
-          params: {
-            query: `id==${fundTypeId}`,
+  const fetchFundDetails = useCallback(
+    () => {
+      setIsLoading(true);
+      mutator.fund.GET().then(response => {
+        setCompositeFund(response);
+        const {
+          fund: {
+            ledgerId,
+            fundTypeId,
+            allocatedFromIds,
+            allocatedToIds,
           },
-        });
-      }
+        } = response;
 
-      if (ledgerId) {
-        mutator.ledger.GET({
-          params: {
-            query: `id==${ledgerId}`,
-          },
-        });
-      }
+        if (fundTypeId) {
+          mutator.fundType.GET({
+            params: {
+              query: `id==${fundTypeId}`,
+            },
+          }).then(fundType => setFundTypeName(fundType[0].name));
+        }
 
-      if (allocatedFromIds.length) {
-        mutator.allocatedFrom.GET({
-          params: {
-            query: allocatedFromIds.map(fundId => `id==${fundId}`).join(' or '),
-          },
-        });
-      }
+        if (ledgerId) {
+          mutator.ledger.GET({
+            params: {
+              query: `id==${ledgerId}`,
+            },
+          }).then(res => setLedger(res[0]));
+        }
 
-      if (allocatedToIds.length) {
-        mutator.allocatedTo.GET({
-          params: {
-            query: allocatedToIds.map(fundId => `id==${fundId}`).join(' or '),
-          },
-        });
-      }
-    });
+        if (allocatedFromIds.length) {
+          mutator.allocatedFrom.GET({
+            params: {
+              query: allocatedFromIds.map(fundId => `id==${fundId}`).join(' or '),
+            },
+          }).then(allocatedFrom => setAllocatedFromFunds(allocatedFrom.map(f => f.name).join(', ')));
+        }
+
+        if (allocatedToIds.length) {
+          mutator.allocatedTo.GET({
+            params: {
+              query: allocatedToIds.map(fundId => `id==${fundId}`).join(' or '),
+            },
+          }).then(allocatedTo => setAllocatedToFunds(allocatedTo.map(f => f.name).join(', ')));
+        }
+        setIsLoading(false);
```

Comment:
IMHO, it should be somewhere above. Also I would suggest to think about this flag - basically, loading is still happening when you set it to `false` here